### PR TITLE
weechat-matrix: 0.1.0 -> 0.2.0

### PR DESCRIPTION
**Pull request by emilazy**
###### Motivation for this change

New upstream release. Tested on top of #87610, so probably merge that first.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix [hunk 1]
```diff
@@ -20,15 +20,17 @@ let
...
-  version = "0.1.0";
+  inherit version;
 
   src = fetchFromGitHub {
     owner = "poljar";
     repo = "weechat-matrix";
-    rev = "65a5db7291439b6132e35e8cc09ed901614fabf6";
-    sha256 = "0m3k5vrv5ab1aw1mjd0r8d71anwqzvncvv9v5zx9xp1i188sdm8x";
+    rev = version;
+    hash = "sha256-qsTdF9mGHac4rPs53mgoOElcujicRNXbJ7GsoptWSGc=";
```
Comment (Ekleog) [inline pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix:33]: Unfortunately, this syntax of setting hashes is valid only with versions of nix more recent than the current support level of nixpkgs, so unless I've missed us bumping it it's not possible to merge this as-is :/ 

When you get to changing this to a `sha256 =` hash, feel free to ping me! :)

**Review by Ekleog — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix (hunk 1 above)
Comment (ryantm) [reply pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix:33]: People on #nixos-dev IRC have told me this format is fine now.

**Comment by ryantm** [comment]
Result of `nixpkgs-review pr 87640` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>

File: pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix (hunk 1 above)
Comment (ryantm) [reply pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix:33]: https://logs.nix.samueldr.com/nixos-dev/2020-09-09#3959626

**Pull request merged**